File: Data Structures & Algorithms/binary-tree-diameter/submission-3.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution 
{
    public int diameterOfBinaryTree(TreeNode root) 
    {

        int[] maxDiameter = new int[1];
        height(root, maxDiameter);
        return maxDiameter[0];
    }

    public int height(TreeNode root, int[] maxDiameter) 
    {
        if(root == null)
        {
            return 0;
        }

        int leftHeight = height(root.left, maxDiameter);
        int rightHeight = height(root.right, maxDiameter);

        maxDiameter[0] = Math.max(maxDiameter[0], leftHeight + rightHeight);

        return 1 + Math.max(leftHeight, rightHeight);
        
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        def visit (node, level):
            if not node:
                return
            
            if len(res) <= level:
                res.append([])

            res[level].append(node.val)

            visit(node.left, level + 1)
            visit(node.right, level + 1)

        visit(root, 0)
        return res

        